packages/grafana-data/src/vector/CircularVector.ts

```diff
@@ -11,16 +11,17 @@ interface CircularOptions<T> {
  * Circular vector uses a single buffer to capture a stream of values
  * overwriting the oldest value on add.
  *
- * This supports addting to the 'head' or 'tail' and will grow the buffer
+ * This supports adding to the 'head' or 'tail' and will grow the buffer
  * to match a configured capacity.
  */
-export class CircularVector<T = any> implements MutableVector<T> {
+export class CircularVector<T = any> extends MutableVector<T> {
```

Comment:
For this class's iterator I thought an infinite iterator made the most sense, but maybe someone more familiar with `CircularVector`'s usage could tell me if having it just iterate over `buffer`'s contents would be more useful.

Reply:
That's a good question. I think we use CircularVectot in the same way as any other Vectors. It's used in CircularDataFrame just to have an easy api for having fixed size, circular buffer (for streaming IIRC). I don't think the infinite iterator makes much sense, iterating over a buffer should be enough. @ryantxu any thoughts on this?